assets/aws/main.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/gravitational/kingpin"
+)
+
+func main() {
+	account := kingpin.Flag("aws-account", "AWS Account ID").Short('a').Required().String()
+	regions := kingpin.Flag("regions", "List of AWS regions to get and update AMIs for").Short('r').Required().String()
+	amiType := kingpin.Flag("type", "Type of AMI: 'oss', 'ent', or 'ent-fips'").Short('t').Required().Enum(string(OSS), string(Ent), string(FIPS))
+	version := kingpin.Flag("version", "Teleport version to update AMIs with").Short('v').Required().String()
+	kingpin.Parse()
+
+	ctx := context.Background()
+
+	imageIDs := make(map[string]string)
+
+	for _, region := range strings.Split(*regions, ",") {
+		stub := fmt.Sprintf("gravitational-teleport-ami-%v-%v", *amiType, *version)
+		if *amiType == "ent-fips" {
+			stub = fmt.Sprintf("gravitational-teleport-ami-ent-%v-fips", *version)
+		}
+
+		cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(region))
```

Comment:
Looks like it would work fine with a role, as long as something before it provides the right creds to use. https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/

Reply:
Yep, works exactly the same as the script that used the AWS CLI.